### PR TITLE
Update docker documentation

### LINKED ISSUE #26423: JOSS Review THM - Docker container not working
I try to run the `docker` container by executing:
```
docker run -ti idaholab/moose:latest /bin/bash -c 'cd test; ./run_tests'
```

This fails with:
```
Building and linking hit...
In file included from /opt/moose/framework/contrib/hit/hitpre311.cpp:1168,
                 from /opt/moose/framework/contrib/hit/hit.cpp:6:
/opt/moose/framework/contrib/hit/parse.h:11:10: fatal error: wasphit/HITInterpreter.h: No such file or directory
   11 | #include "wasphit/HITInterpreter.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from /opt/moose/framework/contrib/hit/braceexpr.h:3,
                 from /opt/moose/framework/contrib/hit/lex.cc:7:
/opt/moose/framework/contrib/hit/parse.h:11:10: fatal error: wasphit/HITInterpreter.h: No such file or directory
   11 | #include "wasphit/HITInterpreter.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from /opt/moose/framework/contrib/hit/parse.cc:10:
/opt/moose/framework/contrib/hit/parse.h:11:10: fatal error: wasphit/HITInterpreter.h: No such file or directory
   11 | #include "wasphit/HITInterpreter.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from /opt/moose/framework/contrib/hit/braceexpr.h:3,
                 from /opt/moose/framework/contrib/hit/braceexpr.cc:1:
/opt/moose/framework/contrib/hit/parse.h:11:10: fatal error: wasphit/HITInterpreter.h: No such file or directory
   11 | #include "wasphit/HITInterpreter.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from /opt/moose/framework/contrib/hit/braceexpr.h:3,
                 from /opt/moose/framework/contrib/hit/main.cc:10:
/opt/moose/framework/contrib/hit/parse.h:11:10: fatal error: wasphit/HITInterpreter.h: No such file or directory
   11 | #include "wasphit/HITInterpreter.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [/opt/moose/framework/moose.mk:138: hit] Error 1
Traceback (most recent call last):
  File "/opt/moose/python/pyhit/__init__.py", line 13, in <module>
    import hit
ModuleNotFoundError: No module named 'hit'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/moose/python/pyhit/__init__.py", line 18, in <module>
    import hit
ModuleNotFoundError: No module named 'hit'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./run_tests", line 6, in <module>
    from TestHarness import TestHarness
  File "/opt/moose/python/TestHarness/__init__.py", line 15, in <module>
    from .TestHarness import TestHarness
  File "/opt/moose/python/TestHarness/TestHarness.py", line 20, in <module>
    from FactorySystem.Factory import Factory
  File "/opt/moose/python/FactorySystem/__init__.py", line 13, in <module>
    from .Parser import Parser
  File "/opt/moose/python/FactorySystem/Parser.py", line 13, in <module>
    import pyhit
  File "/opt/moose/python/pyhit/__init__.py", line 22, in <module>
    import hit
ModuleNotFoundError: No module named 'hit'
```

JOSS Review Thread: https://github.com/openjournals/joss-reviews/issues/6146

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #26423

### CODE REVIEW

**Comment by loganharbour** [comment]
@cticenhour this is ready.

It's not extremely detailed, but it's better than the blatantly wrong content we have right now.

File: modules/doc/content/getting_started/installation/docker.md [hunk 1]
```diff
@@ -1,45 +1,48 @@
...
+docker run -it -v projects:/projects idaholab/moose-dev:latest
+cd /projects
+git clone https://github.com/idaholab/moose.git
+cd moose/test
+make -j 4
+./run_tests -j 4
+```
 
-WORKDIR /opt
+If you are building another image from this one, you may need to source the file at `/environment` first to obtain the proper environment.
```
Comment (cticenhour) [inline modules/doc/content/getting_started/installation/docker.md:25]: Let's make this a note alert to make it stand out.

File: modules/doc/content/getting_started/installation/docker.md [hunk 2]
```diff
@@ -1,45 +1,48 @@
...
 ```
+docker volume create projects
+docker run -it -v projects:/projects idaholab/moose:latest
+cd /projects
+moose-opt --copy-inputs electromagnetics
+cd combined/electromagnetics
+moose-opt --run -j 4
+```
+
+This imaged is versioned based on the master hash of MOOSE that it was compiled with.
```
Comment (cticenhour) [inline modules/doc/content/getting_started/installation/docker.md:48]:
```suggestion
This image is versioned based on the master hash of MOOSE that it was compiled with.
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
Only two items, pending a look at the render.

**Review by cticenhour — APPROVED** [review]

**Pull request merged**